include/aws/auth/private/aws_http_credentials_provider.h

```diff
@@ -0,0 +1,145 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#ifndef AWS_HTTP_CREDENTIALS_PROVIDER_H
+#define AWS_HTTP_CREDENTIALS_PROVIDER_H
+
+#include <aws/auth/credentials.h>
+#include <aws/http/connection_manager.h>
+#include <aws/http/request_response.h>
+
+AWS_PUSH_SANE_WARNING_LEVEL
+
+/**
+ * Configuration needed to create a underlying http client.
+ */
+struct aws_credentials_provider_http_options {
+    struct aws_credentials_provider_shutdown_options shutdown_options;
+    /*
+     * Connection bootstrap to use for any network connections
+     * made while sourcing credentials.
+     * Required.
```

Comment:
trivial: let's put required to the beginning of the doc. So, it's more clear which ones are required, and which ones are optional.